data-products/src/freestar_revenue_reporting/freestar_reporting_flow.py

```diff
@@ -0,0 +1,277 @@
+    headers = {
+        "Authorization": f"Bearer {FREESTAR_API_KEY}",
+        "Content-Type": "application/json"
+    }
+
+    # Create the subdirectory (if it exists, remove it first to ensure it's empty)
+    if os.path.exists(subdirectory):
+        shutil.rmtree(subdirectory)
+        os.makedirs(subdirectory)
+
+    # Define a function to save data to a file
+    def save_data_to_file(data, filename):
+        filepath = os.path.join(subdirectory, filename)
+        with open(filepath, "w") as output_file:
+            json.dump(data, output_file, indent=4)
+
+    # Define a function to save combined data to a Parquet file
+    def save_data_to_parquet(data, filename):
+        filepath = os.path.join(subdirectory, filename)
+        df = pd.DataFrame(data)
+        table = pa.Table.from_pandas(df)
+        pq.write_table(table, filepath)
+
+    page_number = 1  # Initialize the page number
+
+    while True:
+        try:
+            # Send the POST request to the API with the current pagination parameters
+            response = requests.post(API_BASE_URL + API_ENDPOINT, json=request_payload, headers=headers)
+            response.raise_for_status()  # Raise an exception for HTTP errors
+
+            # Parse the JSON response
+            response_json = response.json()
+
+            # Remove the 'NdrPrebid' prefix from keys in the JSON response
+            for item in response_json['data']:
+                for key in list(item.keys()):
+                    if key.startswith('NdrPrebid.'):
+                        new_key = key[len('NdrPrebid.'):]
+                        item[new_key] = item.pop(key)
+
+            # Save the JSON response to a separate file with a page number
+            filename = f"api_response_page_{page_number}.json"
+            save_data_to_file(response_json, filename)
+            print(f"Page {page_number} saved to {filename}")
+
+            # Check if there are more records to retrieve
+            if len(response_json['data']) < 2000:
+                break  # Stop if there are no more records
+
+            # Increment the offset for the next page
+            request_payload["query"]["offset"] += 2000
+            page_number += 1
+
+            # Add a 5-second delay before fetching the next page
+            time.sleep(5)
+
+        except requests.exceptions.RequestException as e:
+            print(f"Error: {e}")
+
+        except json.JSONDecodeError as e:
+            print(f"Error decoding JSON response: {e}")
+
+    # Combine paginated files
+
+    # Loop through all JSON files in the directory
+    for filename in os.listdir(subdirectory):
+        if filename.endswith(".json"):
+            file_path = os.path.join(subdirectory, filename)
+            with open(file_path, "r") as json_file:
+                try:
+                    data = json.load(json_file)["data"]
+                    combined_data.extend(data)  # Extend the list with data blocks
```

Comment:
@mmiermans ha I beat you to that and pushed those changes already :) - not sure what I was thinking in storing the files separately. i think initially it was for testing purposes and we thought it would just be one json.